akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala

```diff
@@ -985,6 +955,123 @@ trait FlowOps[+Out, +Mat] {
   def log(name: String, extract: Out ⇒ Any = _identity)(implicit log: LoggingAdapter = null): Repr[Out, Mat] =
     andThen(Stages.Log(name, extract.asInstanceOf[Any ⇒ Any], Option(log)))
 
+  /**
+   * Combine the elements of current flow and given [[Source]] into a stream of tuples.
+   *
+   * '''Emits when''' all of the inputs has an element available
+   *
+   * '''Backpressures when''' downstream backpressures
+   *
+   * '''Completes when''' any upstream completes
+   *
+   * '''Cancels when''' downstream cancels
+   */
+  def zip[U](right: Graph[SourceShape[U], _]): Repr[(Out, U), Mat] = zipMat(right)(Keep.left)
```

Comment:
Is there a good reason for naming it `right` for scaladsl and `second` for javadsl?